.github/workflows/check-code-quality.yml

```diff
@@ -31,3 +31,32 @@ jobs:
       uses: ludeeus/action-shellcheck@master
       with:
         severity: style
+
+    - name: Check for broken links
+      run: | 
+        touch valid_urls.txt redirected_urls.txt invalid_urls.txt
+        find . -type f \( -name "*.txt" -o -name "*.md" -o -name "*.py" -o -name "*.rst" \) | 
+          xargs grep -Eo "\b(https?|ftp):\/\/[-A-Za-z0-9+&@#\/%?=~_|$!:,.;]*[A-Za-z0-9+&@#\/%=~_|$]" | 
+          grep -v 'drive.google.com' |    # Returns 401 (needs google auth)
+          grep -v 'pipeline-hemis' |      # Returns 403 (private GH repo)
+          grep -v 'sciencedirect.com' |   # Returns 403
+          grep -v 'wiley.com' |           # Returns 403
+          grep -v 'sagepub.com' |         # Returns 403
+          grep -v 'ncbi.nlm.nih.gov' |    # Returns 403          
+          grep -v 'oxfordjournals.org' |  # Returns 403
+          grep -v 'docker.com' |          # Returns 403
+          grep -v 'ieeexplore.ieee.org' | # Returns 418
+          grep -v 'CHANGES.md' |          # Has too many GitHub links, triggers "Too many requests"
```

Comment:
This will match both links in the file `CHANGES.md`, and links where this appears in the URL, is this intended?

If it's just the file that we want to avoid, it could be filtered out at the `find` stage, with an extra `\( \! -name 'CHANGES.md' \)`.